### PR TITLE
MC-178 add explicit datadog connection spec for each function

### PR DESCRIPTION
**Pull request by smaant**
Removed receiving datadog agent host/port through env var or java option and make it an explicit parameter for each function.

### CODE REVIEW

File: test/clj_datadog/core_test.clj [hunk 1]
```diff
@@ -9,63 +9,61 @@
   [& body]
   `(let [sended# (atom nil)]
      (with-redefs
-       [dd/send-msg (fn [value#] (reset! sended# value#))]
+       [dd/send-msg (fn [_# value#] (reset! sended# value#))]
        ~@body
        @sended#)))
 
-(deftest should-format-tags
```
Comment (Otann) [inline test/clj_datadog/core_test.clj]: А почему ты решил убрать `deftest` и заменить на комментарий?

**Comment by Otann** [comment]
А чем это лучше чем переменные окрудения, то есть зачем так делать?

File: test/clj_datadog/core_test.clj (hunk 1 above)
Comment (Otann) [inline test/clj_datadog/core_test.clj]: For future reference and out of plain interest, why you decided to remove structural grouping for tests and replaced it with plain text comments?

Comment (smaant) [inline test/clj_datadog/core_test.clj]: One testing framework nested into another one is not playing well with existed tools.
* `lein test` surprisingly somehow works
* `lein test-refresh` - pretty popular [plugin for lein](https://github.com/jakemcc/lein-test-refresh) doesn't, and even worse, just saying `0 failures, 0 errors`
* `lein expectations` - [official plugin for expectations](https://github.com/gar3thjon3s/lein-expectations) doesn't work either
* `lein autoexpect` - [another official plugin](https://github.com/jakemcc/lein-autoexpect) doesn't work too
* emacs cider test runner - doesn't know how to handle it either

So it's probably the best to use only one test framework.

Comment (Otann) [inline test/clj_datadog/core_test.clj]: Good! Thanks for the explanation.

File: README.md [hunk 2]
```diff
@@ -28,27 +31,38 @@ Then import datadog in your namespace:
 
 Following datadog metric methods are available:
 
+### DataDog connection
+
+You have to provide a map with host and port of DataDog agent or an empty map
+to use default (`{:host "127.0.0.1" :port 8125}`).
+
+You may also create macroses like
```
Comment (DmitryKorolev) [inline README.md]: typo: macroses -> macros

Comment (smaant) [inline README.md]: поправил

**Comment by smaant** [comment]
@DmitryKorolev is it good to merge?

**Pull request merged**